lib/portlayer/storage/vsphere/toolbox_datasource.go

```diff
@@ -39,30 +40,86 @@ func (t *ToolboxDataSource) Source() interface{} {
 
 // Export reads data from the associated data source and returns it as a tar archive
 func (t *ToolboxDataSource) Export(op trace.Operation, spec *archive.FilterSpec, data bool) (io.ReadCloser, error) {
-	defer trace.End(trace.Begin(""))
+	defer trace.End(trace.Begin("toolbox export"))
 
-	// set up file manager
-	client := t.VM.Session.Client.Client
-	filemgr, err := guest.NewOperationsManager(client, t.VM.Reference()).FileManager(op)
+	client, err := GetToolboxClient(op, t.VM, t.ID)
 	if err != nil {
+		op.Errorf("Cannot get toolbox client: %s", err.Error())
 		return nil, err
 	}
 
-	// authenticate client and parse container host/port
-	auth := types.NamePasswordAuthentication{
-		Username: t.ID,
+	var readers []io.Reader
+	for inclusion := range spec.Inclusions {
```

Comment:
@morris-jason I'm not sure this should be a loop - looks like the tether side will handle a spec with multiple inclusions and we're not stripping the spec before passing it through.
Will this result in multiple copies of the same file?